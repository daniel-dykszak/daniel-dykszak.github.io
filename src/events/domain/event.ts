
export type EventLocation = {
    lat: string;
    lng: string;
    address: string;
}
export type Event = {
    id: string;
    title: string;
    description: string;
    category: string;
    location: EventLocation
}