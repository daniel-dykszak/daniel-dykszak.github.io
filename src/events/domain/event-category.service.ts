import {Event} from "./event";

export const getCategoryFromEvents = (events: Event[]) => {
    const removeDuplicates = (el: string, index: number, arr: string[]) =>
        arr.indexOf(el) === index

    return events.map(event => event.category)
        .filter(removeDuplicates)
}