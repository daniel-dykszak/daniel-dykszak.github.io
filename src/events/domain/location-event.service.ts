import {Event} from "./event";

export const findEventById = (list: Event[], eventId: string): Event | undefined => {
    return list.find(e => e.id === eventId);
}

export const getEventSlug = (event: Event): string => {
    const titleForUrl = event.title.replace(/\s/g, "-")
    return `${event.id}-${titleForUrl}`
}

export const getIdFromSlug = (slug: string): string => {
    return slug.split("-")[0];
}

export const filterEventsByCategory = (
    events: Event[],
    category: string,
    favoritesIds: Set<string>): Event[] => {
    return events.filter((event) => {
        if (!category) return true;
        if (category === "favorites") return favoritesIds.has(event.id);
        return event.category === category;
    });
}
