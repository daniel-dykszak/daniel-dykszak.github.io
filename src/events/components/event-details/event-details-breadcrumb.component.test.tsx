import {render} from '@testing-library/react';
import {Event} from "../../domain/event";
import {EventDetailsBreadcrumb} from "./event-details-breadcrumb.component";
import {MemoryRouter} from "react-router-dom";

describe("EventDetailsBreadcrumb component", () => {
    it("should match snapshot", () => {
        const event = {
            category: "someCategory",
            title: "someTitle"
        } as Event;
        const component = render(<MemoryRouter>
            <EventDetailsBreadcrumb event={event}/>
        </MemoryRouter>, {}).baseElement;
        expect(component).toMatchSnapshot();
    });
})