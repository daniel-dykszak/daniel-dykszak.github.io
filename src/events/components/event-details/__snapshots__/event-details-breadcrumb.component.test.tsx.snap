// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetailsBreadcrumb component should match snapshot 1`] = `
<body>
  <div>
    <nav
      aria-label="breadcrumb"
    >
      <ol
        class="breadcrumb"
      >
        <li
          class="breadcrumb-item"
        >
          <a
            href="/"
            title="Home"
          >
            Home
          </a>
        </li>
        <li
          class="breadcrumb-item"
        >
          <a
            href="/event-category/someCategory"
            title="Category someCategory"
          >
            someCategory
          </a>
        </li>
        <li
          aria-current="page"
          class="breadcrumb-item active"
        >
          someTitle
        </li>
      </ol>
    </nav>
  </div>
</body>
`;
